packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.js

```diff
@@ -51,56 +51,59 @@ const SwitchContainer = styled({
   justifyContent: 'center',
 })(PaddedView);
 
-const AddPrayerForm = memo(({ avatarSource, title, btnLabel, ...props }) => (
-  <Formik
-    initialValues={{ prayer: '', anonymous: false }}
-    onSubmit={(values) => console.log(values)}
-  >
-    {({ handleChange, handleBlur, handleSubmit, values }) => (
-      <ModalView {...props}>
-        <FlexedSafeAreaView forceInset={{ top: 'always', bottom: 'never' }}>
-          <ShrinkingView behavior={'padding'}>
-            <HeaderView>
-              {/* TODO: margin will be fixed pending core #667 */}
-              <Avatar
-                source={values.anonymous ? null : avatarSource}
-                size={'medium'}
-              />
-              <TitleView>
-                <H3>{title}</H3>
-              </TitleView>
-            </HeaderView>
-            <FlexedView>
-              <PaddedView>
-                <StyledTextInput
-                  editable
-                  multiline
-                  placeholder="Start typing your prayer..."
-                  onChangeText={handleChange('prayer')}
-                  onBlur={handleBlur('prayer')}
-                  value={values.prayer}
-                  underline={false}
+const AddPrayerForm = memo(
+  ({ onSubmit, avatarSource, title, btnLabel, ...props }) => (
+    <Formik
+      initialValues={{ prayer: '', anonymous: false }}
+      onSubmit={onSubmit}
+    >
+      {({ handleChange, handleBlur, handleSubmit, values }) => (
+        <ModalView {...props}>
+          <FlexedSafeAreaView forceInset={{ top: 'always', bottom: 'never' }}>
+            <ShrinkingView behavior={'padding'}>
+              <HeaderView>
+                {/* TODO: margin will be fixed pending core #667 */}
+                <Avatar
+                  source={values.anonymous ? null : avatarSource}
```

Comment:
```suggestion
                  source={values.anonymous && values ? null : avatarSource}
```

This protects against value being undefined.

Reply:
Don't think values would ever be `undefined` as they are coming from `Formik` so it'll always be something.

Reply:
Cool.